code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -51,8 +53,16 @@
 	set_light(2, 3, l_color = SSticker.cultdat ? SSticker.cultdat.construct_glow : LIGHT_COLOR_BLOOD_MAGIC)
 
 /mob/living/simple_animal/hostile/construct/death(gibbed)
-	. = ..()
 	SSticker.mode.remove_cultist(show_message = FALSE, target_mob = src)
+	if(held_body) // Null check for empty bodies
+		held_body.forceMove(get_turf(src))
+		SSticker.mode.add_cult_immunity(held_body)
+		held_body.key = key
```

Comment:
there';s no .key var on brains, if they're brain sacced they runtime on destroy and cannot die